Recurrent doc fixes

skaae opened the pull request:
- Fix string representation of function in recurent layers
- add Gate to docs
- add Gate to recurrent docstring
- Change RecurrentLayer init to be normal(0.1) like all other layers
- add high forget gate bias example to gate class

craffel commented on lasagne/layers/recurrent.py:
```diff
@@ -379,8 +404,8 @@ class RecurrentLayer(CustomRecurrentLayer):
     def __init__(self, incoming, num_units,
-                 W_in_to_hid=init.Uniform(),
-                 W_hid_to_hid=init.Uniform(),
+                 W_in_to_hid=init.Normal(0.1),
```
Can we address this in a different issue/PR?  I think being consistent is good, but we are also not being consistent with the rest of `lasagne` if we do this.

skaae replied:
yes.

craffel commented on lasagne/layers/recurrent.py:
```diff
@@ -438,6 +468,24 @@ class Gate(object):
         The nonlinearity that is applied to the input gate activation. If None
         is provided, no nonlinearity will be applied.
+
+    References
```
Let's use a different reference; this was actually proposed in the original forget gate paper: http://cognet.mit.edu/system/cogfiles/journalpdfs/089976600300015015.pdf . Can we also move the references under the example?

skaae replied:
It was the only reference I could remember. Can you write the name of the reference? My ip is blocked?

craffel replied:
Sorry, my mistake for browsing refs while on Columbia's network.  The paper is "Learning to forget: continual prediction with LSTM" by Gers.  Using an initial forget gate bias of 5 is mentioned on page 15.

craffel commented:
Thanks, didn't realize layers had started to have their signature repeated like this.

skaae commented:
I changed the reference and moved it + use Uniform init again.

btw have you tried one of the conv test failing randomly? 

```
lasagne/tests/test_theano_extensions.py::test_conv[2-conv1d_sc] FAILED
```

It failed on my first run, but when i ran the tests again it passed?

craffel commented on lasagne/layers/recurrent.py:
```diff
@@ -321,7 +338,15 @@ def step_masked(input_n, mask_n, hid_previous, *args):
-    """Dense recurrent neural network (RNN) layer
+    """
+    lasagne.layers.recurrent.RecurrentLayer(incoming, num_units,
+    W_in_to_hid=init.Uniform(), W_hid_to_hid=init.Uniform(),
```
Shouldn't these be `lasagne.init.Uniform()`?

skaae replied:
Yes. I'll rebase. 

Shouldnt i squash everything into a single commit?

craffel commented:
I just made a small docstring update and rebased against upstream/master, so you'll need to do a rebase.  Sorry!

skaae committed:
Function descriptions and gate docs

skaae commented:
rebased, squashed and fixed the error you spotted.

skaae commented:
I was a little too enthusiastic when cleaning up branches..

craffel commented:
Hm, it's a little weird that when you compile the docs the "References" are forced to show up above the "Examples" section, but I suppose there's not a lot we can do about that.  Merged in d695a394b22e854e40f37d156b41b5696cbdf375, thanks!
